go/arrow/flight/flightsql/server.go

```diff
@@ -640,7 +668,17 @@ type Server interface {
 	// EndTransaction commits or rollsback a transaction
 	EndTransaction(context.Context, ActionEndTransactionRequest) error
 	// CancelQuery attempts to explicitly cancel a query
+	// Deprecated: Since 13.0.0. If you can require all clients
+	// use 13.0.0 or later, you can use only CancelFlightInfo and
+	// you don't need to use CancelQuery. Otherwise, you may need
+	// to use CancelQuery and/or CancelFlightInfo.
```

Comment:
Having `flightSqlServer` have its own implementation of `CancelQuery` as I mentioned above also has the nice effect here of allowing users to switch entirely to `CancelFlightInfo` safely regardless of the client as the server will do the right thing then.

Reply:
You're right.
The document is wrong. Users need only implement `CancelFlightInfo` and it works with `CancelQuery` and `CancelFlightInfo` from clients.